localstack/utils/docker.py

```diff
@@ -596,5 +803,559 @@ def mountable_tmp_file():
+        :return: Tuple (path_exists, path_is_directory)
+        """
+        # Docker CLI copy uses go FileMode to determine if target is a dict or not
+        # https://github.com/docker/cli/blob/e3dfc2426e51776a3263cab67fbba753dd3adaa9/cli/command/container/cp.go#L260
+        # The isDir Bit is the most significant bit in the 32bit struct:
+        # https://golang.org/src/os/types.go?s=2650:2683
+        stats = {}
+        try:
+            _, stats = container.get_archive(container_path)
+            target_exists = True
+        except APIError:
+            target_exists = False
+        target_is_dir = target_exists and bool(stats["mode"] & SDK_ISDIR)
+        return target_exists, target_is_dir
+
+    def get_container_status(self, container_name: str) -> DockerContainerStatus:
+        # LOG.debug("Getting container status for container: %s", container_name) #  too verbose
+        try:
+            container = self.client().containers.get(container_name)
+            if container.status == "running":
+                return DockerContainerStatus.UP
+            else:
+                return DockerContainerStatus.DOWN
+        except NotFound:
+            return DockerContainerStatus.NON_EXISTENT
+        except APIError:
+            raise ContainerException()
+
+    def get_network(self, container_name: str) -> str:
+        LOG.debug("Getting network type for container: %s", container_name)
+        try:
+            container = self.client().containers.get(container_name)
+            return container.attrs["HostConfig"]["NetworkMode"]
+        except NotFound:
+            raise NoSuchContainer(container_name)
+        except APIError:
+            raise ContainerException()
+
+    def stop_container(self, container_name: str) -> None:
+        LOG.debug("Stopping container: %s", container_name)
+        try:
+            container = self.client().containers.get(container_name)
+            container.stop(timeout=0)
+        except NotFound:
+            raise NoSuchContainer(container_name)
+        except APIError:
+            raise ContainerException()
+
+    def remove_container(self, container_name: str, force=True, check_existence=False) -> None:
+        LOG.debug("Removing container: %s", container_name)
+        if check_existence and container_name not in self.get_running_container_names():
+            LOG.debug("Aborting removing due to check_existence check")
+            return
+        try:
+            container = self.client().containers.get(container_name)
+            container.remove(force=force)
+        except NotFound:
+            raise NoSuchContainer(container_name)
+        except APIError:
+            raise ContainerException()
+
+    def list_containers(self, filter: Union[List[str], str, None] = None, all=True) -> List[dict]:
+        if filter:
+            filter = [filter] if isinstance(filter, str) else filter
+            filter = dict([f.split("=") for f in filter])
```

Comment:
nit: Unlikely that the string will contain multiple equals signs (`=`), but to be on the safe side:
```
filter = dict([f.split("=", 1) for f in filter])
```